spec/models/solidus_bolt/bolt_configuration_spec.rb

```diff
@@ -0,0 +1,60 @@
+require 'spec_helper'
+
+RSpec.describe SolidusBolt::BoltConfiguration, type: :model do
+  let(:column_list) {
+    [
+      'id',
+      'bearer_token',
+      'environment_url',
+      'merchant_public_id',
+      'merchant_id',
+      'api_key',
+      'signing_secret',
+      'publishable_key',
+      'created_at',
+      'updated_at'
+    ]
+  }
+
+  describe 'Check Model Integrity' do
+    it 'has correct table name' do
+      expect(described_class.table_name).to eq 'solidus_bolt_bolt_configurations'
+    end
+
+    it 'has correct attributes' do
+      expect(described_class.new.attributes.keys.sort).to eq column_list.sort
+    end
+  end
+
+  describe '#fetch' do
+    it 'fetches the correct record' do
+      bolt_configuration = create(:bolt_configuration)
+      expect(described_class.fetch).to eq(bolt_configuration)
+    end
+  end
+
+  describe '#can_create?' do
+    it 'returns true when no records are present' do
+      expect(described_class).to be_can_create
+    end
+
+    it 'returns false when records are present' do
+      create(:bolt_configuration)
+      expect(described_class).not_to be_can_create
+    end
+  end
+
+  describe 'validations' do
+    it 'adds an error when records are present' do
+      create(:bolt_configuration)
+
+      expect { create(:bolt_configuration) }.to raise_error ActiveRecord::RecordInvalid
```

Comment:
I think you should also have to test that the error is related to the `:base` and that contains the specific error message `Can create only one record for this Model`. 

ref https://relishapp.com/rspec/rspec-rails/v/2-14/docs/model-specs/errors-on

Reply:
The link mentions the use of `errors_on` which is not available on rspec 3.

I have tested the conditions mentioned but in a different way. I hope that should be fine
check here [365a6d4](https://github.com/nebulab/solidus_bolt/commit/365a6d43e7e8a049df560bde89c04183e14bda7a)

Reply:
Yes, it works for me!